src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/WindowsHelpers.cs

```diff
@@ -97,99 +99,122 @@ public static byte[] ValueAsAscii(this Oid oid)
                     throw Marshal.GetLastWin32Error().ToCryptographicException();
                 }
 
-                byte* decoded = stackalloc byte[cb];
+                int MaxStackAllocSize = 256;
+                Span<byte> decoded = stackalloc byte[MaxStackAllocSize];
 
-                if (!Interop.crypt32.CryptDecodeObjectPointer(
-                    Interop.Crypt32.CertEncodingType.All,
-                    lpszStructType,
-                    encoded,
-                    encoded.Length,
-                    Interop.Crypt32.CryptDecodeObjectFlags.None,
-                    decoded,
-                    ref cb))
+                if ((uint)cb > MaxStackAllocSize)
                 {
-                    throw Marshal.GetLastWin32Error().ToCryptographicException();
+                    decoded = new byte[cb];
```

Comment:
Is it possible cb is negative?

Reply:
It's extremely unlikely as the data we are dealing with here is typically a dozen to a few hundred bytes. If it overflows to negative then the array-new will give a runtime exception, but I don't anticipate it would ever happen unless someone was doing something extremely nonsensical.

Reply:
And in case it's ambiguous, Win32 calls it DWORD, so there's not "negative", just "really, really big"